requests/utils.py

```diff
@@ -166,13 +166,14 @@ def get_netrc_auth(url, raise_errors=False):
         netrc_path = None
 
         for f in NETRC_FILES:
-            try:
-                loc = os.path.expanduser('~/{0}'.format(f))
-            except KeyError:
-                # os.path.expanduser can fail when $HOME is undefined and
-                # getpwuid fails. See http://bugs.python.org/issue20164 &
-                # https://github.com/requests/requests/issues/1846
-                return
+            if f.startswith('~'):
```

Comment:
This is not right: we’ll now prepend a tilde to a path we know has one. Given that `expanduser` is safe to call on paths without user flags, we can just unconditionally call it. No need for this change.